app/services/interface_cleaner.rb

```diff
@@ -0,0 +1,51 @@
+class InterfaceCleaner
+  include ActiveRecord::Sanitization
+
+  ESCAPE_CHAR = '\\'
+
+  def self.nic_pattern_to_like(nic)
+    sanitize_sql_like(nic, ESCAPE_CHAR).tr('*', '%')
+  end
+
+  def self.ignored_interface_like_patterns
+    Setting[:ignored_interface_identifiers].map { |pattern| nic_pattern_to_like(pattern) }
+  end
+
+  delegate :logger, :to => :Rails
+
+  attr_reader :primary_nics, :provision_nics, :primary_hosts, :provision_hosts
```

Comment:
No, these lists are for the user - ids in these lists are interfaces that match the ignored pattern, but were not removed, since they are marked as primary and provision. I also show the hosts that have such interfaces.

Reply:
It looks like nics are only used in tests, hosts are used in rake task as well. I am okay with this.